### PR TITLE
feat: Search api examples

### PR DESCRIPTION
**Pull request by adilansari**

### CODE REVIEW

File: src/main/java/com/tigrisdata/starter/models/SearchRequest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package com.tigrisdata.starter.models;
+
+import java.util.List;
+
+public final class SearchRequest {
```
Comment (himank) [inline src/main/java/com/tigrisdata/starter/models/SearchRequest.java:5]: should we add a setter/getter for the filter as well?

File: src/main/java/com/tigrisdata/starter/models/SearchRequest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  /**
+   * List of fields to generate facets for
+   *
+   * @param facetFields List<String>
+   */
+  public void setFacetFields(List<String> facetFields) {
+    this.facetFields = facetFields;
+  }
+
+  public SearchRequest() {
```
Comment (JigarJoshi) [inline src/main/java/com/tigrisdata/starter/models/SearchRequest.java:48]: nit: remove this.

File: src/main/java/com/tigrisdata/starter/models/SearchRequest.java (hunk 1 above)
Comment (JigarJoshi) [inline src/main/java/com/tigrisdata/starter/models/SearchRequest.java:5]: nit: add equals()/hashcode()

File: pom.xml [hunk 3]
```diff
@@ -26,11 +26,11 @@
     </dependencies>
     <properties>
         <!-- we are still pre-release -->
-        <tigris.client.java.version>1.0.0-alpha.17</tigris.client.java.version>
+        <tigris.client.java.version>1.0.0-alpha.22</tigris.client.java.version>
```
Comment (JigarJoshi) [inline pom.xml:29]: update the version of the project to `1.0.0-alpha.22`

File: src/main/java/com/tigrisdata/starter/models/SearchRequest.java (hunk 1 above)
Comment (adilansari) [reply src/main/java/com/tigrisdata/starter/models/SearchRequest.java:5]: Filter can be a little tricky to implement, because we are using swagger as playground interface, user won't know what to put in there. For example, we could use filter in following ways:

```java
public final class SearchRequest {
  private String q;
  private List<String> searchFields;
  private List<String> facetFields;
  private TigrisFilter filter;
}
```

```java
public final class SearchRequest {
  private String q;
  private List<String> searchFields;
  private List<String> facetFields;
  private String filter;
}
```

Swagger UI won't give anything useful to the user to try out `filter` usage (unless we write a custom parser):
Screen Shot 2022-08-05 at 10 04 35 AM


tldr; we can keep Swagger UI simple and easy to understand. And our in-code example demonstrates integration details.

Comment (adilansari) [reply src/main/java/com/tigrisdata/starter/models/SearchRequest.java:5]: Let me know how you feel, I can explore enriching this swagger definition.

**Review by himank — APPROVED** [review]
